content/graph/LCT.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Author: 
+ * Description: link-cut Tree. Supports BBST-like 
+ * augmentations. (Can be used in place of HLD).
+ * Current implementation supports update value at a node,
+ * and query max on a path.
+ * Tested on: http://acm.timus.ru/problem.aspx?num=1553
+ * Status: Passes existing fuzz tests (with function names modified).
+ */
+struct Node {
+	bool flip = 0;
+	Node *pp, *p, *c[2];
+	// add stuff
+	int val, cval;
+	Node() { 
+		pp = p = c[0] = c[1] = 0; 
+		val = cval = 0;
+	}
+	// lazy propogation
+	void push() {
+		if (flip) {
+			if(c[0]) c[0]->flip ^= 1;
+			if(c[1]) c[1]->flip ^= 1;
+			swap(c[0], c[1]); flip = 0;
+		}
+		// add stuff
+	}
+	// combine values from children with self.
+	void pull() {
+		cval = val;
+		if(c[0]) c[0]->push(), cval = max(cval, c[0]->cval); 
+		if(c[1]) c[1]->push(), cval = max(cval, c[1]->cval);
+		// add stuff
+	}
+	void rot(bool t) { /// start-hash
+		Node *y = p, *z = y->p;
+		if (z) z->c[z->c[1] == y] = this;
+		p = z;
+		y->c[t] = c[!t];
+		if (c[!t]) c[!t]->p = y;
+		c[!t] = y;
+		y->p = this;
+		if (z) z->pull();
+		y->pull();
+	} /// end-hash
+	void xiao() {
```

Comment:
I agree that we should probably rename this

Reply:
"propogatePathParent" is a little much though... Will keep thinking about possible alternatives.

Reply:
could certainly shorten that to `propPP` if that's what we want to get across. But a nonsensical name would be fine as well